### DISCUSSION TITLE
Can we get AppSync (GRAPHQL) into the mix of this solution

### DISCUSSION THREAD

**Discussion by siva-nadella** [aws-samples/aws-saas-factory-ref-solution-serverless-saas#46] [Ideas]
Hi Anubhav, I happened to watch your video presenting this few days back and thanks for presenting.

One question I have is, do you have any ideas as to bring AppSync into this mix without disturbing overall architecture of this.

Appreciate your thought.

Thanks

**Comment by anubhavynr** [comment]
You should be able to implement APIs using AppSync. 

AppSync has lambda authorizer that can be used in similar way: 
https://aws.amazon.com/blogs/mobile/appsync-lambda-auth/

Do you plan to use unit resolver or pipeline Resolver?

If using VTL with unit resolver, you can get tenant id from JWT using lambda auth and use that in your VTL. You can also use the new util.log to log tenant id in Cloud Watch.

Lambda resolver will work exactly the way I described in this repo

**Reply by siva-nadella** [reply]
Thanks Anubav for your quick response.

Basically looking for GRAPHKL service over multi tenant SASS with respective isolations supported by a single table design dynamoDB.

Looking to integrate your solution with GRAPHKL as well as single table design.

After checking this https://github.com/alexdebrie/graphql-dynamodb-single-table/blob/main/lib/single-table-cdk-stack.ts

Brainstorming to connect these two ( 1. Your code that creates the entire stack, 2. Single Table DynamoDB GRAPHKL : https://github.com/alexdebrie/graphql-dynamodb-single-table/blob/main/lib/single-table-cdk-stack.ts )

Per your question, at this point make this work with unit resolver, but I do have needs to connect to other third part http services.

Appreciate your time engaging this conversation.

Thanks,
Siva
